internal/migrations/observability/api/convert.go

```diff
@@ -0,0 +1,113 @@
+package api
+
+import (
+	logging "github.com/openshift/cluster-logging-operator/api/logging/v1"
+	obs "github.com/openshift/cluster-logging-operator/api/observability/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	obsruntime "github.com/openshift/cluster-logging-operator/internal/runtime/observability"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ConvertLoggingToObservability converts a logging.ClusterLogging and/or logging.ClusterLogForwarder
+// to observability.ClusterLogForwarder
+func ConvertLoggingToObservability(k8sClient client.Client, loggingCL *logging.ClusterLogging, loggingCLF *logging.ClusterLogForwarder, outputSecrets map[string]*corev1.Secret) *obs.ClusterLogForwarder {
+	// Legacy Workflow
+	if loggingCL != nil && loggingCL.Name == constants.SingletonName && loggingCL.Namespace == constants.OpenshiftNS {
+		clToObsCLF := obsruntime.NewClusterLogForwarder(loggingCL.Namespace, loggingCL.Name, runtime.Initialize)
+		// Legacy ClusterLogging only
+		if loggingCLF == nil {
+			obsClfSpec := convertLegacyClusterLogging(&loggingCL.Spec)
+			clToObsCLF.Spec = *obsClfSpec
+			return clToObsCLF
+		}
+		// Legacy ClusterLogging + ClusterLogForwarder
+		clToObsCLF.Spec = convertClusterLogForwarder(loggingCL, loggingCLF, outputSecrets, true)
+		return clToObsCLF
+	}
+
+	// custom clf
+	obsCLF := obsruntime.NewClusterLogForwarder(loggingCLF.Namespace, loggingCLF.Name, runtime.Initialize)
+	obsCLF.Spec = convertClusterLogForwarder(loggingCL, loggingCLF, outputSecrets, false)
+
+	return obsCLF
+}
+
+// convertLegacyClusterLogging generates the output and pipelines determined by the logstore type of the ClusterLoggingInstance
+func convertLegacyClusterLogging(loggingCLSpec *logging.ClusterLoggingSpec) *obs.ClusterLogForwarderSpec {
+	obsDefaultOut := *generateDefaultOutput(loggingCLSpec.LogStore)
+
+	var obsCollSpecs *obs.CollectorSpec
+	if loggingCLSpec.Collection != nil {
+		obsCollSpecs = &obs.CollectorSpec{
```

Comment:
@Clee2691 we need to cover the usecase of someone using the old collector semantics where 'spec.logs' takes precedence over 'spec.collector'.  This comes out of a discussion with support